tests/unit/test_core.py

```diff
@@ -0,0 +1,171 @@
+import pytest
+from west import core
+from west import exceptions
+from west import settings
+
+
+@pytest.fixture
+def known_versions(mocker):
+    versions = ["1.1", "1.2", "1.3"]
+    mocker.patch("west.core.files.get_known_versions", return_value=versions)
+
+    return versions
+
+
+def test_get_applied_versions(mocker, known_versions):
+    mocker.patch("west.core.db.get_applied_versions", return_value=["1.0", "1.1"])
```

Comment:
Do we plan to have functional tests for functions in the db module ?

Reply:
I think we should test it, but no easy to do in a pure functional way